react/src/components/UserCredentialList.tsx

```diff
@@ -0,0 +1,518 @@
+  transformSorterToOrderString,
+} from '../helper';
+import { useUpdatableState } from '../hooks';
+import { useBAIPaginationOptionState } from '../hooks/reactPaginationQueryOptions';
+import BAIPropertyFilter from './BAIPropertyFilter';
+import BAITable from './BAITable';
+import Flex from './Flex';
+import { UserCredentialListDeleteMutation } from './__generated__/UserCredentialListDeleteMutation.graphql';
+import { UserCredentialListModifyMutation } from './__generated__/UserCredentialListModifyMutation.graphql';
+import { UserCredentialListQuery } from './__generated__/UserCredentialListQuery.graphql';
+import {
+  DeleteOutlined,
+  InfoCircleOutlined,
+  LoadingOutlined,
+  ReloadOutlined,
+  SettingOutlined,
+} from '@ant-design/icons';
+import {
+  App,
+  Button,
+  Popconfirm,
+  Radio,
+  Tag,
+  Tooltip,
+  Typography,
+  theme,
+} from 'antd';
+import graphql from 'babel-plugin-relay/macro';
+import dayjs from 'dayjs';
+import _ from 'lodash';
+import { BanIcon, UndoIcon } from 'lucide-react';
+import { useState, useTransition } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLazyLoadQuery, useMutation } from 'react-relay';
+
+const UserCredentialList: React.FC = () => {
+  const { t } = useTranslation();
+  const { token } = theme.useToken();
+  const { message } = App.useApp();
+
+  const [fetchKey, updateFetchKey] = useUpdatableState('first');
+  const [activeType, setActiveType] = useState<'active' | 'inactive'>('active');
+  const [order, setOrder] = useState<string | undefined>('-created_at');
+  const [filterString, setFilterString] = useState<string>();
+  const [isPendingRefresh, startRefreshTransition] = useTransition();
+  const [isActiveTypePending, startActiveTypeTransition] = useTransition();
+  const [isPendingPageChange, startPageChangeTransition] = useTransition();
+  const [isPendingFilter, startFilterTransition] = useTransition();
+
+  const {
+    baiPaginationOption,
+    tablePaginationOption,
+    setTablePaginationOption,
+  } = useBAIPaginationOptionState({
+    current: 1,
+    pageSize: 20,
+  });
+
+  const { keypair_list } = useLazyLoadQuery<UserCredentialListQuery>(
+    graphql`
+      query UserCredentialListQuery(
+        $limit: Int!
+        $offset: Int!
+        $filter: String
+        $order: String
+        $domain_name: String
+        $email: String
+        $is_active: Boolean
+      ) {
+        keypair_list(
+          limit: $limit
+          offset: $offset
+          filter: $filter
+          order: $order
+          domain_name: $domain_name
+          email: $email
+          is_active: $is_active
+        ) {
+          items {
+            id
+            user_id
+            access_key
+            is_admin
+            resource_policy
+            created_at
+            rate_limit
+            num_queries
+            concurrency_used @since(version: "24.09.0")
+            concurrency_limit @since(version: "24.09.0")
+          }
```

Comment:
Is `concurrency_used` added in version 24.09.0? In the schema, `concurrency_limit` has a deprecated directive. Please check the version compatibility and modify if needed.

Reply:
`concurrency_used` is added in version 24.09.0 but, `concurrency_limit` is moved to `KeyPairResourcePolicy` object as `max_concurrent_session`. 

Since the schema doesn't list the exact version that is deprecated, and that information is inaccurate, I'll start by writing another query to get an additional KeyPairResourcePolicy for `max_concurrent_session`.

And I will request that the `resource_policy` field provided in the `KeyPair` object be provided as a `KeyPairResourcePolicy` instead of a string value.

@yomybaby

Reply:
Based on our discussions,I will not show them in that list, but rather on the Resource Policy page.

[teams](https://teams.microsoft.com/l/message/19:14c484402d874dafb15806d093b95a82@thread.skype/1734342656931?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1733797630870&teamName=devops&channelName=Frontend&createdTime=1734342656931)